controllers/providers/aws/autoscaling.go

```diff
@@ -54,17 +54,23 @@ func (a *AmazonClientSet) TerminateInstance(instance *autoscaling.Instance) erro
 	return nil
 }
 
-func (a *AmazonClientSet) SetInstanceStandBy(instance *autoscaling.Instance, scalingGroupName string) error {
-	instanceID := aws.StringValue(instance.InstanceId)
-	input := &autoscaling.EnterStandbyInput{
-		AutoScalingGroupName:           aws.String(scalingGroupName),
-		InstanceIds:                    aws.StringSlice([]string{instanceID}),
-		ShouldDecrementDesiredCapacity: aws.Bool(false),
+func (a *AmazonClientSet) SetInstanceStandBy(instances []*autoscaling.Instance, scalingGroupName string) (bool, error) {
+	var instanceIDs []string
+	for _, instance := range instances {
+		if aws.StringValue(instance.LifecycleState) == autoscaling.LifecycleStateInService {
+			instanceIDs = append(instanceIDs, aws.StringValue(instance.InstanceId))
+		}
 	}
-
-	if _, err := a.AsgClient.EnterStandby(input); err != nil {
-		return err
+	if len(instanceIDs) > 0 {
+		input := &autoscaling.EnterStandbyInput{
+			AutoScalingGroupName:           aws.String(scalingGroupName),
+			InstanceIds:                    aws.StringSlice(instanceIDs),
+			ShouldDecrementDesiredCapacity: aws.Bool(false),
+		}
+		if _, err := a.AsgClient.EnterStandby(input); err != nil {
+			return true, err
+		}
 	}
 
-	return nil
+	return false, nil
```

Comment:
shouldn't the condition be switched? e.g. true = an instance was set to standby, false = there is an error or no instances were set

Reply:
Oops. Thanks for catching it.

Reply:
My bad. It should be this:

```
func (a *AmazonClientSet) SetInstanceStandBy(instances []*autoscaling.Instance, scalingGroupName string) (bool, error) {
	var instanceIDs []string
	for _, instance := range instances {
		if aws.StringValue(instance.LifecycleState) == autoscaling.LifecycleStateInService {
			instanceIDs = append(instanceIDs, aws.StringValue(instance.InstanceId))
		}
	}
	if len(instanceIDs) > 0 {
		input := &autoscaling.EnterStandbyInput{
			AutoScalingGroupName:           aws.String(scalingGroupName),
			InstanceIds:                    aws.StringSlice(instanceIDs),
			ShouldDecrementDesiredCapacity: aws.Bool(false),
		}
		if _, err := a.AsgClient.EnterStandby(input); err != nil {
			return false, err
		}
		return true, nil
	}

	return false, nil
}
```

Reply:
There could be 3 return conditions.